File: Team04_API/Team04_API/Models/Ticket/Ticket_Updates.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using Team04_API.Models.FAQ;
using Team04_API.Models.Users;

namespace Team04_API.Models.Ticket
{
    public class Ticket_Updates
    {
        [Key]
        public int Ticket_Update_ID { get; set; }

        //[ForeignKey(nameof(ticket))]
        public int Ticket_ID { get; set; }

        //[ForeignKey(nameof(ticket_status))]
        public int? Ticket_Status_Old_ID { get; set; }

        //[ForeignKey(nameof(ticket_status))]
        public int? Ticket_Status_New_ID { get; set; }

        public DateTime DateOfChange { get; set; }

        public bool hasBeenDismissed { get; set; }


        // Virtual 
        public virtual Ticket? ticket { get; set; }

        public virtual Ticket_Status? ticket_status { get; set; }
        //public virtual Ticket_Status ticket_status_new { get; set; }
    }
}

File: Team04_Angular/src/app/Classes/ticket-updates.ts
import { User } from "./users-classes";

export class TicketUpdates {
    ticket_Update_ID: number = 0;
    ticket_ID: number = 0;
    ticket_Status_Old_ID: number = 0;
    ticket_Status_New_ID: number = 0;
    dateOfChange: Date = new Date();
    hasBeenDismissed: boolean = false;
    ticket_Description: string = ''; 
    client: User = new User(); 
}
